lib/app/admin/data-reset.js

```diff
@@ -0,0 +1,212 @@
+      .then(function() {
+        eachSeriesCb();
+      }).catch(function(err) {
+        deferred.reject(err);
+      });
+  }, function(err) {
+    if (err) {
+      deferred.reject(err);
+    } else {
+      deferred.resolve();
+    }
+  });
+
+  return deferred.promise;
+}
+
+/**
+ * Deletes all items of a type - gets list of items based on their type name and removes synchronously one-by-one
+ * @param {Object} mediator - mediator object used for communication
+ * @param {String} itemType - type of the item to remove e.g. result, workorder
+ * @returns {Promise} Promise, resolved if all items are successfully deleted.
+ */
+function deleteAllItems(mediator, itemType) {
+  var deferred = q.defer();
+  var topic = 'wfm:cloud:' + itemType + ':list';
+
+  //get list of item type
+  mediator.request(topic)
+    .then(function(list) {
+      //for each item in the list - mediator delete
+      deleteFromList(mediator, list, itemType, function(err) {
+        if (err) {
+          deferred.reject(err);
+        }
+        deferred.resolve();
+      });
+    }).catch(function(err) {
+      deferred.reject(err);
+    });
+
+  return deferred.promise;
+}
+
+
+/**
+ * Syncrounsly published mediator delete topic for each item on the list
+ * @param {Object} mediator  - mediator object used for communication
+ * @param {Array} list - list of the items to be deleted
+ * @param {String} topic - topic prefix to be published
+ * @param {Callback} loopCb - async.eachSeries callback that is being called after loop iterated through the whole list
+ */
+function deleteFromList(mediator, list, itemType, loopCb) {
+  async.eachSeries(list, function(item, eachLoopCb) {
+    var topic = 'wfm:cloud:' + itemType + ':delete';
+    mediator.request(topic, item.id)
+      .then(function() {
+        eachLoopCb();
+      })
+      .catch(function(err) {
+        //not rejecting here, try to remove others
+        console.log('err in data-reset: item ' + item.id + err);
+        return new Error(err);
+      });
+  }, loopCb);
+}
+
+/**
+ * Builds data using seed data located in ./data-demo, synchronously publishes 'wfm:cloud:{itemType}:create topic for each item.
+ * @param {Object} mediator  - mediator object used for communication
+ * @param {Array} list - list of the items to be used for building
+ * @returns {Promise} Promise, resolved if all items are successfully created.
+ */
+function buildData(mediator, list) {
+  var deferred = q.defer();
+  var relationsMap = [];
+  async.eachSeries(list, function(itemType, eachTypeCb) {
+    relationsMap[itemType] = [];
+
+    var topic = 'wfm:cloud:' + itemType;
+    var demoData = require('./data-demo/' + itemType);
+    async.eachSeries(demoData(), function(dataItem, cb) {
+      var mapping = {
+        seedId: dataItem.id,
+        newId: ''
+      };
+      dataItem = mapRelations(itemType, dataItem, relationsMap[DEFAULT_LIST_RELATIONS[itemType]]);
+      mediator.request(topic + ':create', dataItem)
```

Comment:
`mediator.request` needs a unique identifier to work correctly (e.g. https://github.com/feedhenry-raincatcher/raincatcher-sync/blob/master/lib/server.js#L26)